lib/proxy/plugin_validators.rb

```diff
@@ -7,6 +7,7 @@ def initialize(plugin, setting_name)
 
     def validate!
       setting_value = @plugin.settings.send(@setting_name)
+      return true if setting_value.nil?
```

Comment:
Yes, I like the presence/validation idea.  The two things it wouldn't permit are:

1. A required value without a default.  I guess in these cases we have to put *something* just as an example, even if it isn't a useful default.
2. An optional value with a default.  This just becomes an example in the configuration file.

That seems acceptable, and I agree that it would be logical from a user and developer's perspective.

It might be worth a quick scout through the plugins (http://projects.theforeman.org/projects/foreman/wiki/List_of_Smart-Proxy_Plugins) to see if there are any default config files or settings that this might not work for.